File: skinmate(추천해조)/auth/src/main/java/com/skinmate/auth/domain/Member.java
package com.skinmate.auth.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "member")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Member {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Integer memberId;
    
    @Column(name = "oauth_provider")
    private String oauthProvider;
    
    @Column(name = "oauth_id")
    private String oauthId;
    
    @Column(name = "name")
    private String name;
    
    @Column(name = "email")
    private String email;
    
    @Column(name = "role")
    private String role;
    
    @Column(name = "skin_type")
    private String skinType;
    
    // @Column(name = "min_price")
    // private Integer minPrice;
    
    // @Column(name = "max_price")
    // private Integer maxPrice;
    
    @Column(name = "gender")
    private String gender;
    
    @Column(name = "age_group")
    private Integer ageGroup;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "created_id")
    private Integer createdId;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @Column(name = "updated_id")
    private Integer updatedId;
}

File: skinmate(추천해조)/backend/app/models/member.py
from sqlalchemy import Column, Integer, String
from .common import Base, Common

class Member(Base, Common):
    __tablename__ = "member"

    member_id = Column(Integer, primary_key=True, autoincrement=True)

    oauth_provider = Column(String(50), nullable=True)  # google, kakao, naver
    oauth_id = Column(String(100), nullable=True)

    name = Column(String(100), nullable=True)
    email = Column(String(100), nullable=True)
    role = Column(String(20), nullable=True, default="USER")

    skin_type = Column(String(50), nullable=True)           # 건성, 지성, 복합성, 민감성 등
    gender = Column(String(10), nullable=True)              # 성별
    age_group = Column(Integer, nullable=True)           # 나이대

